File: EcmApiRestService/RestService/Models/FileKeySet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Agilent.ECMAPI;

namespace EcmapiRestService.Models
{
    public class FileKeySet
    {
        public int Index;
        public string KeyType;
        public string Filter;
        public List<EcmapiRestService.Models.FileKey> filekeys = new List<EcmapiRestService.Models.FileKey>();

        public FileKeySet() { }

        public FileKeySet(Agilent.ECMAPI.FileKeySet fileKeySet)
        {
            this.Index = fileKeySet.Index;
            this.KeyType = fileKeySet.KeyType;
            this.Filter = fileKeySet.Filter;
            foreach (Agilent.ECMAPI.FileKey fileKey in fileKeySet)
            {
                filekeys.Add(new EcmapiRestService.Models.FileKey(fileKey));
            }
        }
    }
}

File: EcmApiClient/src/com/pfizer/ecmapi/client/models/FileKeySet.java
package com.pfizer.ecmapi.client.models;

import java.util.ArrayList;
import java.util.List;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("FileKeySet")
public class FileKeySet {
	@XStreamAlias("Index")
    private int index;
	@XStreamAlias("KeyType")
	private String keyType;
	@XStreamAlias("Filter")
    private String filter;
    private List<FileKey> filekeys = new ArrayList<FileKey>();

    public FileKeySet() { }

	public int getIndex() {
		return index;
	}

	public void setIndex(int index) {
		this.index = index;
	}

	public String getKeyType() {
		return keyType;
	}

	public void setKeyType(String keyType) {
		this.keyType = keyType;
	}

	public String getFilter() {
		return filter;
	}

	public void setFilter(String filter) {
		this.filter = filter;
	}

	public List<FileKey> getFilekeys() {
		return filekeys;
	}

	public void setFilekeys(List<FileKey> filekeys) {
		this.filekeys = filekeys;
	}

}
